tasks/task_3/transfer_learning.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+"""Transfer Learning
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/11W4P9OMoLafNSM5TWLVsR095Bgxe7t2e
+"""
+
+from google.colab import drive
+drive.mount('/content/gdrive')
+
+PATH_TO_DATA = "/content/gdrive/MyDrive/data" 
+!ls {PATH_TO_DATA}
+
+import numpy as np 
+import pandas as pd 
+import os
+import cv2
+
+import matplotlib.pyplot as plt
+import tensorflow as tf
+from tensorflow.keras import layers
+from tensorflow.keras.preprocessing import image_dataset_from_directory
+
+BATCH_SIZE = 32
+IMG_SIZE = (160,160)
+
+train_dataset = image_dataset_from_directory(PATH_TO_DATA,
+                                              validation_split=0.3,
+                                              subset= "training",
+                                              seed=42,
+                                              image_size= IMG_SIZE,
+                                              batch_size=BATCH_SIZE)
+
+validation_dataset = image_dataset_from_directory(PATH_TO_DATA,
+                                                  validation_split = 0.3,
+                                                  shuffle=True,
+                                                  subset= "validation",
+                                                  seed=42,
+                                                  batch_size=BATCH_SIZE,
+                                                  image_size=IMG_SIZE
+                                                  )
+
+val_batches = tf.data.experimental.cardinality(validation_dataset)
+test_dataset = validation_dataset.take(val_batches // 5)
+validation_dataset = validation_dataset.skip(val_batches // 5)
+
+print('Number of validation batches: %d' % tf.data.experimental.cardinality(validation_dataset))
+print('Number of test batches: %d' % tf.data.experimental.cardinality(test_dataset))
+
+AUTOTUNE = tf.data.AUTOTUNE
+
+train_dataset = train_dataset.prefetch(buffer_size=AUTOTUNE)
+validation_dataset = validation_dataset.prefetch(buffer_size=AUTOTUNE)
+test_dataset = test_dataset.prefetch(buffer_size=AUTOTUNE)
+
+data_augmentation = tf.keras.Sequential([
+  tf.keras.layers.experimental.preprocessing.RandomFlip('horizontal'),
+  tf.keras.layers.experimental.preprocessing.RandomRotation(0.2),
+])
+
+preprocess_input = tf.keras.applications.mobilenet_v2.preprocess_input
+
+rescale = tf.keras.layers.experimental.preprocessing.Rescaling(1./127.5, offset= -1)
+
+IMG_SHAPE = IMG_SIZE + (3,)
+base_model = tf.keras.applications.MobileNetV2(input_shape=IMG_SHAPE,

```

Comment:
If you are using mobilenet it's better to use 224x224 image shape

https://keras.io/api/applications/mobilenet/

But yeah, should work with 160, not an error, just may impact accuracy